books/serializers.py

```diff
@@ -1,24 +1,51 @@
+import re
+
 from rest_framework import serializers
 from django.contrib.auth import get_user_model
+from rest_framework.validators import UniqueValidator
+
+from books.models import Author, Book
+from rest_framework.relations import HyperlinkedIdentityField, HyperlinkedRelatedField
+from django_countries.serializers import CountryFieldMixin
+
+
+class UserSerializer(CountryFieldMixin, serializers.ModelSerializer):
+    class Meta:
+        model = get_user_model()
+        fields = '__all__'
+
+
+class BookSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = Book
+        fields = ('name', 'publication_date',)
+
 
-from books.models import Author
+class AuthorSerializer(serializers.HyperlinkedModelSerializer):
+    url = HyperlinkedIdentityField(view_name='author-detail')
+    # user = UserSerializer(required=True)
+    user = HyperlinkedRelatedField(allow_null=True,
+                                   queryset=get_user_model().objects.all(),
+                                   required=False,
+                                   validators=[UniqueValidator(queryset=Author.objects.all())],
+                                   view_name='customuser-detail')
+    books = BookSerializer(source="book_set", many=True, required=False)
 
+    class Meta:
+        model = Author
+        fields = '__all__'
 
-class AuthorSerializer(serializers.Serializer):
-    short_name = serializers.CharField(max_length=200)
-    area_of_interest = serializers.CharField(max_length=200)
-    user_id = serializers.IntegerField()
+    def validate_short_name(self, short_name):
+        if not re.match(r"[a-zA-Z]", short_name):
+            raise serializers.ValidationError("Short Name must contain only alphabetical characters")
+        else:
+            return short_name
 
     def create(self, validated_data):
-        user_id = validated_data['user_id']
-        short_name = validated_data['short_name']
-        area_of_interest = validated_data['area_of_interest']
-
-        user = get_user_model().objects.get(id=user_id)
-        return Author.objects.create(short_name=short_name, area_of_interest=area_of_interest, user=user)
-
-    def update(self, instance, validated_data):
-        instance.short_name = validated_data.get('short_name', instance.short_name)
-        instance.area_of_interest = validated_data.get('area_of_interest', instance.area_of_interest)
-        instance.save()
-        return instance
+        books = validated_data.get('book_set', None)
+        validated_data.pop('book_set', None)
```

Comment:
- Combine line 45 and 46 i.e.
```suggestion
        books = validated_data.pop('book_set', None)
```